### PR TITLE
add monitor module

### PR DESCRIPTION
**Pull request by zhang1002**
This module's feature:
1. get all SQL's metrics during runtime
2. monitor the sqlOperation
3. for auto-configuring

### CODE REVIEW

File: kyuubi-monitor/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+## for kyuubi-monitor
```
Comment (ulysses-you) [inline kyuubi-monitor/README.md:1]: `##` -> `#`

we can add some simple description in this md. For example:
```
This module is used to monitor kyuubi which include kyuubi server and kyuubi engine.
```

File: kyuubi-monitor/pom.xml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <groupId>org.apache.kyuubi</groupId>
+        <artifactId>kyuubi</artifactId>
+        <version>1.3.0-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>kyuubi-monitor</artifactId>
```
Comment (yaooqinn) [inline kyuubi-monitor/pom.xml:30]: kyuubi-spark-monitor

File: kyuubi-monitor/pom.xml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    <packaging>jar</packaging>
+    <name>Kyuubi Project Monitor</name>
```
Comment (yaooqinn) [inline kyuubi-monitor/pom.xml:32]: Kyuubi Project Spark Monitor

File: pom.xml [hunk 4]
```diff
@@ -35,6 +35,7 @@
         <module>kyuubi-main</module>
         <module>kyuubi-metrics</module>
         <module>kyuubi-zookeeper</module>
+        <module>kyuubi-monitor</module>
```
Comment (yaooqinn) [inline pom.xml:38]: move up to L37

**Review by ulysses-you — APPROVED** [review]

File: pom.xml [hunk 5]
```diff
@@ -28,6 +28,7 @@
         <module>dev/kyuubi-codecov</module>
         <module>externals/kyuubi-download</module>
         <module>externals/kyuubi-spark-sql-engine</module>
+        <module>externals/kyuubi-spark-monitor</module>
```
Comment (pan3793) [inline pom.xml:31]: Sort alphabetically

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merged!

**Pull request closed without merging**